hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/spi/meta/TransactionMetadata.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi.meta;
+
+import com.hedera.services.legacy.core.jproto.JKey;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.Transaction;
+import java.util.List;
+
+/**
+ * Metadata collected when transactions are handled as part of "pre-handle". This happens with
+ * multiple background threads. Any state read or computed as part of this pre-handle, including any
+ * errors, are captured in the TransactionMetadata. This is then made available to the transaction
+ * during the "handle" phase as part of the HandleContext.
+ */
+public interface TransactionMetadata {
+    /**
+     * Checks the failure by validating the status is not {@link ResponseCodeEnum OK}
+     *
+     * @return returns true if status is not OK
+     */
+    boolean failed();
+
+    /**
+     * Returns the failure {@link ResponseCodeEnum}, which gives more information on the failure
+     *
+     * @return response code of the failure
+     */
+    ResponseCodeEnum failureStatus();
+
+    /**
+     * Transaction that is being pre-handled
+     *
+     * @return transaction that is being pre-handled
+     */
+    Transaction getTxn();
```

Comment:
As mentioned in other parts of the review, I'm not sure `Transaction` is the right type here. Maybe it is, but `Transaction` is not really meaningful to us because we need to the specific `TransactionBody` data type. This needs some more thought.

Reply:
Changed to `TransactionBody` type in `TransactionMetadata`.